app/controllers/api/v1/classrooms_controller.rb

```diff
@@ -1,92 +1,101 @@
+        @classroom = Classroom.new(classroom_params)
+        @classroom.user = current_user
+        authorize @classroom
+        if @classroom.save
+          render :show
+        else
+          render_error
+        end
+      end
 
-  def destroy
-    @classroom.destroy
-    render json: {}
-  end
+      def destroy
+        @classroom.destroy
+        render json: {}
+      end
 
-  def initiate_all_work_groups
-    work_groups = @classroom.work_groups.where(aasm_state: :created)
-    work_groups.update_all(aasm_state: :in_progress)
-    render json: @classroom.work_groups.where(aasm_state: :in_progress).to_json
-  end
+      def initiate_all_work_groups
+        work_groups = @classroom.work_groups.where(aasm_state: :created)
+        work_groups.update_all(aasm_state: :in_progress)
+        render json: @classroom.work_groups.where(aasm_state: :in_progress).to_json
+      end
 
-  def conclude_all_work_groups
-    work_groups = @classroom.work_groups.where(aasm_state: :in_progress)
-    # need to make sure that all the worksheets are updated before sending the emails
-    mail_worksheets_to_teacher(work_groups)
-    work_groups.each { |work_group| mail_worksheets(work_group) }
-    work_groups.update_all(aasm_state: :done)
-    render json: @classroom.work_groups.where(aasm_state: :done).to_json
-  end
+      def conclude_all_work_groups
+        work_groups = @classroom.work_groups.where(aasm_state: :in_progress)
+        # need to make sure that all the worksheets are updated before sending the emails
+        mail_worksheets_to_teacher(work_groups)
+        work_groups.each { |work_group| mail_worksheets(work_group) }
+        work_groups.update_all(aasm_state: :done)
+        render json: @classroom.work_groups.where(aasm_state: :done).to_json
+      end
 
-  private
+      private
 
-  def mail_worksheets(work_group)
-    DemoMailer.with(
-      students: work_group.students.to_a,
-      student_group: work_group,
-      image_urls: work_group.worksheets.pluck(:image_url)
-    ).send_worksheets.deliver_later
-    work_group.update(worksheet_email_sent: true)
-  end
+      def mail_worksheets(work_group)
+        DemoMailer.with(
+          students: work_group.students.to_a,
+          student_group: work_group,
+          image_urls: work_group.worksheets.pluck(:image_url)
+        ).send_worksheets.deliver_later
+        work_group.update(worksheet_email_sent: true)
+      end
 
-  def mail_worksheets_to_teacher(work_groups)
-    DemoMailer.with(
-      teacher: work_groups.first.classroom.teacher,
-      work_groups: work_groups.deep_pluck(users: %i[name pen_colour], worksheets: :image_url)
-    ).send_worksheets_to_teacher.deliver_later
-  end
+      def mail_worksheets_to_teacher(work_groups)
+        DemoMailer.with(
+          teacher: work_groups.first.classroom.teacher,
+          work_groups: work_groups.map do |w|
+                         Hash[:users, w.students.pluck(:name, :pen_colour).map do |name, pen_colour|
```

Comment:
have you tested this? `pen_colour` should be under StudentsWorkGroup now so I don't think you can pluck it from students

Reply:
Yes, I tested this! So when I run it, the SQL query does an inner join with users and student_work_group so that's how it's able to get the pen_colour with the name

Reply:
ok, lets keep it then